test/libethereum/blockchain.cpp

```diff
@@ -848,4 +849,17 @@ BOOST_AUTO_TEST_CASE(userDefinedFile)
 	dev::test::userDefinedTest(dev::test::doBlockchainTests);
 }
 
+BOOST_AUTO_TEST_CASE(bcMining)
+{
+	dev::TransientDirectory m_td;				//BlockChain Dir
+	dev::TransientDirectory m_td2;				//State Dir
+
+	CanonBlockChain<BasicAuthority> m_blockchain(m_td.path(), WithExisting::Kill);
+	OverlayDB stateDB = State::openDB(m_td2.path(), m_blockchain.genesisHash());
+	State m_state (m_blockchain.genesisState(stateDB));
+
+	m_state.sync(m_blockchain);
+	mine(m_state, m_blockchain);
```

Comment:
Dont you need some BOOST_REQUIRE to check for successful mining? You only test that no exceptions are thrown here.